src/js/pwm.js

```diff
@@ -66,29 +66,21 @@ function PWM(pin, options, callback) {
   // Check arguments.
   if (util.isFunction(options)) {
     callback = options;
-    self._options.dutyCycle = -1;
-    self._options.period = -1;
   } else if (util.isObject(options)) {
-    if (util.isNumber(options.dutyCycle)) {
-      self._options.dutyCycle = options.dutyCycle;
-    } else {
-      self._options.dutyCycle = -1;
-    }
-
-    if (util.isNumber(options.period)) {
-      self._options.period = options.period;
-    } else {
-      self._options.dutyCycle = -1;
+    var dutyCycle = options.dutyCycle;
+    var period = options.period;
+    if (!util.isNumber(period) && util.isNumber(options.frequency)) {
+      period = 1.0 / options.frequency;
```

Comment:
It can be divided by 0. If you do not check it elsewhere, please handle the error.